src/phases/query/CollScanLargeNumberOfFields.yml

```diff
@@ -0,0 +1,359 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  This workload tests the performance of collection scan queries against a collection containing
+  documents with a large number of fields.
+
+Keywords:
+- Loader
+- CrudActor
+- QuiesceActor
+- insert
+- find
+
+GlobalDefaults:
+  Database: &Database {^Parameter: {Name: Database, Default: unused}}
+  DocumentCount: &DocumentCount {^Parameter: {Name: DocumentCount, Default: -1}}
+  Repeat: &Repeat {^Parameter: {Name: Repeat, Default: -1}}
+  Collection: &Collection Collection0
+  MaxPhases: &MaxPhases 18
+
+  # Generates a random string of 5-50 characters.
+  RandomShortString: &RandomShortString {^FastRandomString: {
+    length: {^RandomInt: {min: 5, max: 50}}
+  }}
+  # Generates a random string of 50-500 characters.
+  RandomLongString: &RandomLongString {^FastRandomString: {
+    length: {^RandomInt: {min: 50, max: 500}}
+  }}
+  # Generates a random UUID.
+  RandomUUID: &RandomUUID {^UUID: {hex: {^FormatString: {format: "%08x-%04x-%04x-%04x-%012x",
+                                withArgs: [
+                                  {^RandomInt: {min: 0, max: 0xFFFFFFFF}},
+                                  {^RandomInt: {min: 0, max: 0xFFFF}},
+                                  {^RandomInt: {min: 0, max: 0xFFFF}},
+                                  {^RandomInt: {min: 0, max: 0xFFFF}},
+                                  {^RandomInt: {min: 0, max: 0xFFFFFFFFFFFF}}]}}}}  
+
+  FieldNames: &FieldNames [
+    "bool",
+    "int",
+    "double",
+    "string",
+    "objectId",
+    "uuid",
+    "date",
+    "object",
+    "array",
+    "anytype",
+  ]
+  FieldValues: &FieldValues [
+    &RandomBool {^Choose: {from: [true, false, null]}},
+    &RandomInt {^RandomInt: {min: 0, max: 1_000_000_000}},
+    &RandomDouble {^RandomDouble: {min: 0, max: 1_000_000_000}},
+    &RandomString {^Choose: {from: [*RandomShortString, *RandomLongString, null]}},
+    &RandomObjectId {^ObjectId: {^RandomString: {length: 24, alphabet: "0123456789ABCDEF"}}},
+    *RandomUUID,
+    &RandomDate {^RandomDate: {min: "1980-01-01", max: "2025-01-01"}},
+    &RandomObject {^Object: {
```

Comment:
Does this mean we will only have documents with one level of nesting?

Reply:
Yep, all RandomObjects only have one level of nesting. Overall, we have paths with up to 4 levels (e.g. `arr.a.object1.randomfield`).